doc/developer-notes.md

```diff
@@ -941,7 +941,9 @@ Threads and synchronization
     internal to a class (private or protected) rather than public.
 
   - Combine annotations in function declarations with run-time asserts in
-    function definitions:
+    function definitions (`AssertLockNotHeld()` can be omitted if `LOCK()` is
+    called unconditionally after it because `LOCK()` does the same check as
```

Comment:
> ... can be omitted if \<condition\>

It creates additional burden for reviewing if \<condition\> will change in the future, doesn't it?

Reply:
If you're moving locks around (`<condition>` changes), then you should be reviewing whether `AssertLockNotHeld` is still appropriate either way?

Reply:
Yes, if `LOCK()` is moved, then it would have to be reviewed either way.

Reply:
Hoe to merge